src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/WinGetPackageManagerCommand.cs

```diff
@@ -88,18 +89,113 @@ public void RepairUsingLatest(bool preRelease, bool allUsers, bool force)
             var runningTask = this.RunOnMTA(
                 async () =>
                 {
+                    if (!string.IsNullOrWhiteSpace(expectedVersion))
+                    {
+                        this.Write(StreamType.Verbose, $"Attempting to resolve version '{expectedVersion}'");
+                        expectedVersion = await this.ResolveVersionAsync(expectedVersion, includePrerelease);
+                    }
+                    else
+                    {
+                        this.Write(StreamType.Verbose, "No version specified.");
+                    }
+
                     await this.RepairStateMachineAsync(expectedVersion, allUsers, force);
                     return true;
                 });
             this.Wait(runningTask);
         }
 
+        /// <summary>
+        /// Tries to get the latest version matching the pattern.
+        /// </summary>
+        /// <remarks>
+        /// Pattern only supports trailing wildcards.
+        /// - For example, the pattern can be: 1.11.*, 1.11.3*
+        /// - But it cannot be: 1.*1.1 or 1.*1*.1.
+        /// </remarks>
+        /// <param name="versions">List of versions to match against.</param>
+        /// <param name="pattern">Pattern to match.</param>
+        /// <param name="includePrerelease">Include prerelease versions.</param>
+        /// <param name="result">The resulting version.</param>
+        /// <returns>True if a matching version was found.</returns>
+        private static bool TryGetLatestMatchingVersion(IEnumerable<WinGetVersion> versions, string pattern, bool includePrerelease, out WinGetVersion result)
+        {
+            pattern = string.IsNullOrWhiteSpace(pattern) ? "*" : pattern;
+
+            var parts = pattern.Split('.');
+            string major = parts[0];
+            string minor = parts.Length > 1 ? parts[1] : string.Empty;
+            string build = parts.Length > 2 ? parts[2] : string.Empty;
+            string revision = parts.Length > 3 ? parts[3] : string.Empty;
+
+            if (!includePrerelease)
+            {
+                versions = versions.Where(v => !v.IsPrerelease);
+            }
+
+            versions = versions
+                .Where(v =>
+                    VersionPartMatch(major, v.Version.Major) &&
+                    VersionPartMatch(minor, v.Version.Minor) &&
+                    VersionPartMatch(build, v.Version.Build) &&
+                    VersionPartMatch(revision, v.Version.Revision))
+                .OrderBy(f => f.Version);
+
+            if (!versions.Any())
+            {
+                result = null!;
+                return false;
+            }
+
+            result = versions.Last();
+            return true;
+        }
+
+        private static bool VersionPartMatch(string partPattern, int partValue)
+        {
+            if (string.IsNullOrWhiteSpace(partPattern))
+            {
+                return true;
+            }
+
+            if (partPattern.EndsWith("*"))

```

Comment:
I know you aren’t claiming to support it in this PR, but it seems like it would be trivial to add the case for 
```
if (partPattern.StartsWith("*"))
{
    return partValue.ToString().EndsWith(partPattern.TrimStart('*'));
}
```

?